File: backend/src/models/Campaign.js
const mongoose = require("mongoose");

const CampaignSchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: { type: String },
  budget: { type: Number, required: true },
  startDate: { type: Date, required: true },
  endDate: { type: Date, required: true },
  createdBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
}, { timestamps: true });

module.exports = mongoose.model("Campaign", CampaignSchema);

File: backend/src/models/campaign.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from .user import Base

class Campaign(Base):
    __tablename__ = "campaigns"

    id = Column(Integer, primary_key=True)
    name = Column(String, nullable=False)
    description = Column(String)
    budget = Column(String)
    startDate = Column(String)
    endDate = Column(String)
    createdBy = Column(String)
    user_id = Column(Integer, ForeignKey("users.id"))

    user = relationship("User", backref="campaigns")
